### ISSUE TITLE
Guys please! Any help would be greatly appreciated!! No luck using in Ubuntu

### ISSUE THREAD

**Issue by thetechxperts** [axi0mX/ipwndfu#93]
Hello all and a special thank you to Mr. axi0mX! I am not new to Ubuntu, all though its been a few years. I downloaded and installed on an extra machine I have (perm install, not a VM). My issue is this. Ive tried all commands and always get the reply "bad interpreter: No such file or directory". I've tried a few other commands and still nothing. Maybe im doing something wrong? I right click on the folder and choose "open in terminal". So I know im in the right directory. Any help would greatly be appreciated!!! I've been dying to try this out for days now, but no luck. Thanks all

**Comment by slaesh** [comment]
libusb installed?

`sudo apt-get install libusb-1.0-0-dev`

**Comment by thetechxperts** [comment]
> libusb installed?
> 
> `sudo apt-get install libusb-1.0-0-dev`

Hello brother, and thank you so much for the reply. I have installed libimobiledevice already.
any other suggestions would be greatly appreciated

**Comment by thetechxperts** [comment]
I think I may know. Do I need to move the ipwndfu folder to the root or the "/" of the HDD? Could this be my issue? Folder is currently on desktop, but I don't think this could be the issue as I have open the command line from the folder directly.....

**Comment by EWouters** [comment]
Please do not open an issue for support questions like this. Try to find help elsewhere. In this case [StackOverflow](https://stackoverflow.com/questions/14219092/bash-script-and-bin-bashm-bad-interpreter-no-such-file-or-directory) may have your answer. Please close this issue.

**Comment by thetechxperts** [comment]
> Please do not open an issue for support questions like this. Try to find help elsewhere. In this case [StackOverflow](https://stackoverflow.com/questions/14219092/bash-script-and-bin-bashm-bad-interpreter-no-such-file-or-directory) may have your answer. Please close this issue.

I am staying right here chum. Glad you took the time to type a reply, when it could have taken the same amount of effort for you to try to help. I am here as this is an "issue" that I am trying to resolve. Please don't suggest what I should or should not do, as I could care less about you and what you do. Thanks and good day to you

**Comment by EWouters** [comment]
I asked nicely and took the time to try to help. No need to be offended. Check [here](https://opensource.guide/how-to-contribute/#communicating-effectively) for some pointers on how to ask a question more effectively.

**Comment by thetechxperts** [comment]
> I asked nicely and took the time to try to help. No need to be offended. Check [here](https://opensource.guide/how-to-contribute/#communicating-effectively) for some pointers on how to ask a question more effectively.

Thanks for the suggestions. I'm not offended by anyone at any point. This is just the internet. However, you came off as a "boss" in your initial reply, without offering even an iota of help. You definitely didn't ask nicely. Again, I am here needing help on why none of the commands are working, when my environment is setup properly. Maybe its an Ubuntu thing, as i've just tried this on Mac and its works fine. But I still want to get it to work, as I was told it can operate on Ubuntu. Have a blessed day, and its ok to jump off your high horse once in a while!

**Comment by melvyn2** [comment]
You don't have python installed at `/usr/bin/python`, or it is corrupted.

**Comment by DaJakerBoss** [comment]
Hey there! I happened to stumble across this a bunch myself. First off, I'd like to rectify one thing you said:

> Hello brother, and thank you so much for the reply. I have installed libimobiledevice already. any other suggestions would be greatly appreciated

For this, I sincerely hope you're kidding. `libusb` is completely unrelated to `libimobiledevice` and happen to be, I may note, two separate packages. You need this, and it's literally stated in the README, under the big bold "Dependencies"
> libusb, `If you are using Linux: install libusb using your package manager.`

So, to start, you need to install `libusb-dev`

However we still haven't addressed your Python issue. I thought it was peculiar that python3 was already installed but didn't actually work, so I did the apt install for the standard `python` package (`apt install python`) and was delivered an installation of python2.7. Amazingly, this fixed my issue. I'm still toying with my 5S to make sure that Python 2.7 will support checkm8 (if anyone else in this thread could provide some input here that would be helpful) but end of the day it ran.

In summary, run `sudo apt install python libusb-dev` and please remember to run ipdwn as root (`sudo ./ipwndfu`)

**Comment by elihwyma** [comment]
Yeah that still isn't working

**Comment by fgarcia14** [comment]
First: use python 2, with python 3 not working
Second: `sudo apt install libusb-dev libusb-1.0.0-dev libusb-0.1-4`
Thirt: run with `sudo ./ipwndfu -p `

**Comment by ps4gentoo** [comment]
sudo python2.7 ./ipwndfu ..........
